client/src/assets/components/subcomponents/Register.jsx

```diff
@@ -111,8 +114,65 @@ const Register = () => {
     formData.matchingPassword,
   ]);
 
+  useEffect(() => {
+    if (!usernameFocus || !username) return;
+    console.log("formData.username", username, formData.username);
+    const handler = setTimeout(() => {
+      const checkUsername = async () => {
+        try {
+          const response = await axios.get(`${process.env.NEXT_BACKEND_URL}/api/users/username/${formData.username}`);
+          if (response.data) {
+            // Username exists -> not available
+            setUsernameAvailable(false);
+          }
+        } catch (error) {
+          if (error.response && error.response.status === 404) {
+            setUsernameAvailable(true);
+          } else {
+            console.error('Error checking username:', error);
+          }
+        }
+      };
+
+      checkUsername();
+    }, 300); // wait 300ms after last typing
+
+    return () => clearTimeout(handler); // cleanup timer if user keeps typing
+  }, [usernameFocus, formData.username]);
+
+  useEffect(() => {
+    if (!emailFocus || !formData.email) return; // Ensure emailFocus and email are valid
+    
+    const handler = setTimeout(() => {
+      const checkEmail = async () => {
+        try {
+          const response = await axios.get(`${process.env.NEXT_BACKEND_URL}/api/users/email/${formData.email}`);
```

Comment:
Can be shortened without the ${process.env.NEXT_BACKEND_URL} due to the explanation above about the base_url.

Hence reason behind the REGISTER_URL variable.